hikari/users.py

```diff
@@ -120,6 +125,8 @@ class PartialUser(snowflakes.Unique, abc.ABC):
     API call.
     """
 
+    _dm_channel: typing.Optional[channels.DMChannel] = None
```

Comment:
I'm not sure how I feel about this solution, seems to be inconsistent for the interface style approach we're using with the PartialUser here as we're now introducing a variable to a class which is supposedly to solely be an interface. Could we not have send be an abstractmethod where how it's implemented comes down to the implementation's detail? Also would this actually work or would ._dm_channel end up being treated as a read only class attribute on an instance of this object due to lsotting?
(btw this is what i was talking about earlier, was just not in the right place to write this up all afternoon)

Reply:
@FasterSpeeding rewrote it :+1: